libraries/AP_ADSB/AP_ADSB.cpp

```diff
@@ -198,10 +199,15 @@ void AP_ADSB::init(void)
     if (detected_num_instances == 0) {
         for (uint8_t i=0; i<ADSB_MAX_INSTANCES; i++) {
             detect_instance(i);
-            if (_backend[i] != nullptr) {
-                // we loaded a driver for this instance, so it must be
-                // present (although it may not be healthy)
+            if (_backend[i] == nullptr) {
+                continue;
+            }
+
+            if (_backend[i]->init()) {
```

Comment:
I'd have done this the other way around, cleaning up in the case of `!_backend[i]->init()`.  It fits the pattern better here, where we `continue` if the nullptr check triggers.


Not critical.

Reply:
yeah, I actually had it that way at first but it felt like too many continues for a small thing like this.

Reply:
I changed it to:
```
        for (uint8_t i=0; i<ADSB_MAX_INSTANCES; i++) {
            detect_instance(i);
            if (_backend[i] == nullptr) {
                continue;
            }
            if (!_backend[i]->init()) {
                delete _backend[i];
                _backend[i] = nullptr;
                continue;
            }
            // success
            detected_num_instances = i+1;
        }
```